### PR TITLE
Add 3 Arabic classification datasets

### PR DESCRIPTION
**Pull request by mmhamdy**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

This PR adds 3 Arabic datasets for the classification task.
**Note**: The evaluation results of `intfloat/multilingual-e5-base` are included for comparison.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [POINTS.md](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) file.

### CODE REVIEW

File: mteb/tasks/Classification/ara/HotelReviewSentimentClassification.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  publisher={Springer}
+}
+""",
+        n_samples={"train": 2048},
+        avg_character_length={"train": 137.2},
+    )
+
+    def dataset_transform(self):
+        self.dataset["train"] = (
+            self.dataset["train"].shuffle(seed=self.seed).select(range(2048))
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/ara/HotelReviewSentimentClassification.py:46]: Better define a constant at the begining of the script: `N_SAMPLES=2048`, and also use it in the metadata `n_samples={"train": N_SAMPLES}`
```suggestion
            self.dataset["train"].shuffle(seed=self.seed).select(range(N_SAMPLES))
```

File: mteb/tasks/Classification/ara/HotelReviewSentimentClassification.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+@article{elnagar2018hotel,
+  title={Hotel Arabic-reviews dataset construction for sentiment analysis applications},
+  author={Elnagar, Ashraf and Khalifa, Yasmin S and Einea, Anas},
+  journal={Intelligent natural language processing: Trends and applications},
+  pages={35--52},
+  year={2018},
+  publisher={Springer}
+}
+""",
+        n_samples={"train": 2048},
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/ara/HotelReviewSentimentClassification.py:40]:
```suggestion
        n_samples={"train": N_SAMPLES},
```

File: mteb/tasks/Classification/ara/HotelReviewSentimentClassification.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from mteb.abstasks import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/ara/HotelReviewSentimentClassification.py:5]:
```suggestion
N_SAMPLES = 2048

```

File: mteb/tasks/Classification/ara/TweetEmotionClassification.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+}
+""",
+        n_samples={"train": 2048},
+        avg_character_length={"train": 78.8},
+    )
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.rename_column("tweet", "text")
+        self.dataset["train"] = (
+            self.dataset["train"].shuffle(seed=self.seed).select(range(2048))
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/ara/TweetEmotionClassification.py:47]: Same as previously for the number of samples, better use a variable

File: mteb/tasks/Classification/ara/TweetSarcasmClassification.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+}
+""",
+        n_samples={"test": 2110},
+        avg_character_length={"test": 102.1},
+    )
+
+    def dataset_transform(self):
+        # labels: 0 non-sarcastic, 1 sarcastic
+        self.dataset = self.dataset.rename_column("tweet", "text")
+        self.dataset = self.dataset.rename_column("sarcasm", "label")
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/ara/TweetSarcasmClassification.py:59]:
```suggestion
        self.dataset = self.dataset.rename_columns({"tweet": "text", "sarcasm": "label"})
```

**Review by imenelydiaker — CHANGES_REQUESTED** [review]
Nice job! I made some suggestions. Please consider adding contribution points in the POINTS.md file.

File: mteb/tasks/Classification/ara/TweetEmotionClassification.py (hunk 4 above)
Comment (mmhamdy) [reply mteb/tasks/Classification/ara/TweetEmotionClassification.py:47]: Agree. Much better! will address this in the remaining two and add points. Thanks very much for your suggestions 🙂

**Review by imenelydiaker — APPROVED** [review]

**Pull request merged**